### PR TITLE
Optimise inference

### PR DESCRIPTION
**Pull request by randy-seng**
OP#1113

### CODE REVIEW

**Review by martinbaerwolff — CHANGES_REQUESTED** [review]

File: OTVision/detect/detect.py [hunk 1]
```diff
@@ -224,7 +227,7 @@ def write(
     if overwrite or not detections_file_already_exists:
         # Write JSON
         with open(detection_file, "w") as f:
-            json.dump(detections, f, indent=4)
```
[the remark this answers is not in the document]
Comment (randy-seng) [reply OTVision/detect/detect.py:227]: Disabling indentation reduces the size of the otdet file significantly. I ran detection on a 15 min video with the indentation enabled and disabled. The result was that the file size could be reduced by approx. 56% without indentation.

While pretty formatted json is much more readable, the otdet file's purpose is to be processed further (i.e. OTAnalytics).

File: OTVision/detect/detect.py [hunk 2]
```diff
@@ -45,6 +45,7 @@ def main(
     overwrite: bool = CONFIG["DETECT"]["OVERWRITE"],
     ot_labels_enabled: bool = CONFIG["DETECT"]["OTLABELS_ENABLES"],
     debug: bool = CONFIG["DETECT"]["DEBUG"],
+    force_reload_torch_hub_cache: bool = False,
```
[the remark this answers is not in the document]
Comment (randy-seng) [reply OTVision/detect/detect.py]: Done!

**Review by randy-seng — COMMENTED** [review]
Change requests applied except for pretty formatting json output

File: OTVision/detect/detect.py [hunk 3]
```diff
@@ -140,6 +158,7 @@ def main(
         if debug:
             reset_debug()
         return detections_img_file_chunks, class_labels
+    log.info("")
```
Comment (martinbaerwolff) [inline OTVision/detect/detect.py:161]: Tried to guess, but nothing came to my mind. Better you fill the quotation marks :)

Comment (martinbaerwolff) [reply OTVision/detect/detect.py:161]: When done, please merge :) Rest looks good!

Comment (randy-seng) [reply OTVision/detect/detect.py:161]: I forgot to remove that one!

**Review by martinbaerwolff — APPROVED** [review]

**Pull request merged**